src/actix/api_key.rs

```diff
@@ -71,3 +87,22 @@ where
         })
     }
 }
+
+lazy_static! {
+    static ref READ_ONLY_PATHS: Vec<Regex> = vec![
+        Regex::new(r"^\/collections\/[a-z0-9\-\_\&\;]+\/points\/count\/?$").expect("bad regex"),
+        Regex::new(r"^\/collections\/[a-z0-9\-\_\&\;]+\/points\/recommend\/?$").expect("bad regex"),
+        Regex::new(r"^\/collections\/[a-z0-9\-\_\&\;]+\/points\/recommend\/groups\/?$")
+            .expect("bad regex"),
+        Regex::new(r"^\/collections\/[a-z0-9\-\_\&\;]+\/points\/recommend\/batch\/?$")
+            .expect("bad regex"),
+        Regex::new(r"^\/collections\/[a-z0-9\-\_\&\;]+\/points/?$").expect("bad regex"),
+        Regex::new(r"^\/collections\/[a-z0-9\-\_\&\;]+\/points\/scroll\/?$").expect("bad regex"),
+    ];
+}
+
+fn is_read_only(req: &ServiceRequest) -> bool {
```

Comment:
The reported handler name from Actix is not correct here. Calling `ServiceRequest::match_name()` matches by name and not the method so it ends up in cases with the wrong handler. I resolved to matching against regexes manually here. I'm not sure if there's an easier way to figure out the handler.

I added the `lazy_static` dependency to avoid paying the cost of compiling the regex more than once.

Reply:
Nice!